core/equations.gms

```diff
@@ -558,44 +565,76 @@ q_emiEnFuelEx(t,regi,emiTe(enty))..
 ***--------------------------------------------------
 *' Total energy-emissions per emission market, region and timestep  
 ***--------------------------------------------------
-q_emiTeMkt(t,regi,emiTe(enty),emiMkt)..
+q_emiTeMkt(t,regi,emiTe(enty),emiMkt) ..
   vm_emiTeMkt(t,regi,enty,emiMkt)
   =e=
-***   emissions from fuel combustion
+    !! emissions from fuel combustion
```

Comment:
No we do not.
We use them all over the place to have the comments not break up the indentation, preserving readability.
```
$ ag --no-heading -C 1 '^ *!!'
modules/05_initialCap/on/preloop.gms:21-  =e=
modules/05_initialCap/on/preloop.gms:22:    !! Pathway I: FE to ppfEn.
modules/05_initialCap/on/preloop.gms:23-    sum(fe2ppfEn(enty,in),
--
modules/05_initialCap/on/preloop.gms:26-  ) * s05_inic_switch
modules/05_initialCap/on/preloop.gms:27:    !! Pathway II: FE via UE to ppfEn
modules/05_initialCap/on/preloop.gms:28-  + sum(ue2ppfen(enty,in),
--
modules/05_initialCap/on/preloop.gms:31-    ) * s05_inic_switch
modules/05_initialCap/on/preloop.gms:32:    !! Pathway III: FE via ES to ppfEn
modules/05_initialCap/on/preloop.gms:33:    !! For the ES layer, we have to be consistent with conversion and share
modules/05_initialCap/on/preloop.gms:34:    !! parameters when providing FE demands from CES node values.
modules/05_initialCap/on/preloop.gms:35-  + sum(feViaEs2ppfen(enty,in,teEs),
--
modules/05_initialCap/on/preloop.gms:45-    ) * s05_inic_switch
modules/05_initialCap/on/preloop.gms:46:    !! Transformation pathways that consume this enty:
modules/05_initialCap/on/preloop.gms:47-  + sum(en2en(enty,enty2,te),
--
modules/05_initialCap/on/preloop.gms:51-    )
modules/05_initialCap/on/preloop.gms:52:    !! subtract couple production pathways that produce this enty (= add couple production pathways that consume this enty):
modules/05_initialCap/on/preloop.gms:53-  - sum(pc2te(enty3,enty4,te2,enty),
--
modules/05_initialCap/on/preloop.gms:163-
modules/05_initialCap/on/preloop.gms:164:    !! ensure that the production in 2005 is the same as in initialcap2
modules/05_initialCap/on/preloop.gms:165-    loop (pe2se(entyPe,entySe,te),

modules/35_transport/edge_esm/presolve.gms:17-
modules/35_transport/edge_esm/presolve.gms:18:    !! load FE-to-ES results from EDGE-Transport into auxilliary parameter
modules/35_transport/edge_esm/presolve.gms:19-    Execute_Loadpoint "p35_fe2es", p35_fe2es_aux = p35_fe2es;
modules/35_transport/edge_esm/presolve.gms:20:    !! update module parameter with EDGE-Transport results, preserving 2005 data
modules/35_transport/edge_esm/presolve.gms:21:    !! so no altered 2005 data gets passed on to potential fixed runs
modules/35_transport/edge_esm/presolve.gms:22-    p35_fe2es(t,regi,"%cm_GDPscen%","%cm_demScen%","%cm_EDGEtr_scen%",teES_dyn35)$( t.val gt 2005 ) 
modules/35_transport/edge_esm/presolve.gms:23-    = p35_fe2es_aux(t,regi,"%cm_GDPscen%","%cm_demScen%","%cm_EDGEtr_scen%",teES_dyn35);
modules/35_transport/edge_esm/presolve.gms:24:    !! up date model parameter with data after 2010
modules/35_transport/edge_esm/presolve.gms:25-    pm_fe2es(t,regi,teEs_dyn35)$(t.val gt 2010 AND t.val ge cm_startyear AND t.val le 2100)

modules/01_macro/singleSectorGr/equations.gms:93-  =e=
modules/01_macro/singleSectorGr/equations.gms:94:  !! use exp(log(a) * b) = a ** b because the latter is not accurate in GAMS for
modules/01_macro/singleSectorGr/equations.gms:95:  !! very low values of a
modules/01_macro/singleSectorGr/equations.gms:96-  exp(

modules/47_regipol/regiCarbonPrice/presolve.gms:55-        sum(emiBECCS2te(enty,enty2,te,enty3),vm_emiTeDetail.l(t,regi,enty,enty2,te,enty3))
modules/47_regipol/regiCarbonPrice/presolve.gms:56:        !! Industry BECCS (using biofuels in Industry with CCS)
modules/47_regipol/regiCarbonPrice/presolve.gms:57-      + sum((emiMkt,entySe,secInd37,entyFe)$entySeBio(entySe), pm_IndstCO2Captured(t,regi,entySe,entyFe,secInd37,emiMkt))

modules/47_regipol/regiCarbonPrice/datainput.gms:144-
modules/47_regipol/regiCarbonPrice/datainput.gms:145:  !!initialize first and terminal years auxiliary parameters for price targets
modules/47_regipol/regiCarbonPrice/datainput.gms:146-  loop(ttot,  
--
modules/47_regipol/regiCarbonPrice/datainput.gms:178-
modules/47_regipol/regiCarbonPrice/datainput.gms:179:  !!initialize first and terminal years auxiliary parameters for price targets
modules/47_regipol/regiCarbonPrice/datainput.gms:180-  loop(ttot,  

modules/47_regipol/regiCarbonPrice/postsolve.gms:413-        sum(emiBECCS2te(enty,enty2,te,enty3),vm_emiTeDetail.l(t,regi,enty,enty2,te,enty3))
modules/47_regipol/regiCarbonPrice/postsolve.gms:414:        !! Industry BECCS (using biofuels in Industry with CCS)
modules/47_regipol/regiCarbonPrice/postsolve.gms:415-      + sum((emiMkt,entySe,secInd37,entyFe)$entySeBio(entySe), pm_IndstCO2Captured(t,regi,entySe,entyFe,secInd37,emiMkt))
--
modules/47_regipol/regiCarbonPrice/postsolve.gms:443-        sum(emiBECCS2te(enty,enty2,te,enty3),vm_emiTeDetail.l(ttot,regi,enty,enty2,te,enty3))
modules/47_regipol/regiCarbonPrice/postsolve.gms:444:        !! Industry BECCS (using biofuels in Industry with CCS)
modules/47_regipol/regiCarbonPrice/postsolve.gms:445-      + sum((emiMkt,entySe,secInd37,entyFe)$entySeBio(entySe), pm_IndstCO2Captured(ttot,regi,entySe,entyFe,secInd37,emiMkt))
--
modules/47_regipol/regiCarbonPrice/postsolve.gms:525-          * ( 2 * ( exp( -0.025 * iteration.val) + 0.1 ) )  !! in order to also have some dampening over iterations, 
modules/47_regipol/regiCarbonPrice/postsolve.gms:526:      !! this line decreases from 2.1 at iteration 1 to 0.36 in iteration 100. 
modules/47_regipol/regiCarbonPrice/postsolve.gms:527-      )
--
modules/47_regipol/regiCarbonPrice/postsolve.gms:540-            * ( 2 * ( exp( -0.025 * iteration.val) + 0.1 ) )  !! in order to also have some dampening over iterations, 
modules/47_regipol/regiCarbonPrice/postsolve.gms:541:              !! this line decreases from 2.1 at iteration 1 to 0.36 in iteration 100. 
modules/47_regipol/regiCarbonPrice/postsolve.gms:542-          )

modules/80_optimization/nash/postsolve.gms:291-     if (1 le iteration.val,
modules/80_optimization/nash/postsolve.gms:292:        !! no last iteration if this is the first; NA value in p80_repyLastOptim is
modules/80_optimization/nash/postsolve.gms:293:        !! sticky, so test this separately
modules/80_optimization/nash/postsolve.gms:294-        if ( p80_repy(regi,'modelstat') eq 7
modules/80_optimization/nash/postsolve.gms:295:           !! The 1E-4 are quite arbitrary. One should do more research on how
modules/80_optimization/nash/postsolve.gms:296:           !! the solution differs over iteration when status 7 occurs. 
modules/80_optimization/nash/postsolve.gms:297-           AND p80_convNashObjVal_iter(iteration,regi) lt - 1e-4,

modules/21_tax/on/equations.gms:234-  =e=
modules/21_tax/on/equations.gms:235:  !! 1. sustainability tax on production
modules/21_tax/on/equations.gms:236-    v21_tau_bio(t)  * vm_pebiolc_price(t,regi)
modules/21_tax/on/equations.gms:237-    * vm_fuExtr(t,regi,"pebiolc","1")
modules/21_tax/on/equations.gms:238:  !! 2. emission-factor-based tax on consumption
modules/21_tax/on/equations.gms:239-  + p21_bio_EF(t,regi) * pm_taxCO2eq(t,regi)
--
modules/21_tax/on/equations.gms:275-    sum(en2en(enty,enty2,te)$(teFlexTax(te)),
modules/21_tax/on/equations.gms:276:      !! vm_flexAdj is electricity price reduction/increases for flexible/
modules/21_tax/on/equations.gms:277:      !! inflexible technologies change sign such that flexible technologies
modules/21_tax/on/equations.gms:278:      !! get subsidy
modules/21_tax/on/equations.gms:279-      -vm_flexAdj(t,regi,te) 

modules/33_CDR/portfolio/datainput.gms:9-*** direct air capture input data
modules/33_CDR/portfolio/datainput.gms:10:!! Beutler et al. 2019 (Climeworks)
modules/33_CDR/portfolio/datainput.gms:11-p33_fedem("dac", "feels") = 5.28; !! FE demand electricity for ventilation
--
modules/33_CDR/portfolio/datainput.gms:19-;
modules/33_CDR/portfolio/datainput.gms:20:!! display f33_maxProdGradeRegiWeathering;
modules/33_CDR/portfolio/datainput.gms:21:!! $include "./modules/33_CDR/portfolio/input/p33_transport_costs.inc"
modules/33_CDR/portfolio/datainput.gms:22-p33_transport_costs(regi, rlf, rlf) = 0.05; !! KK a temporary fix before p33_transport_costs are moved to mrremind

modules/45_carbonprice/NDC/postsolve.gms:73-        p45_taxCO2eqLastNDCyear(regi) * p45_taxCO2eqYearlyIncrease**(t.val-p45_lastNDCyear(regi)) * (max(p45_taxCO2eqConvergenceYear,t.val) - t.val)
modules/45_carbonprice/NDC/postsolve.gms:74:        !! global, weight going from 0 in NDC target year to 1 in and after 2100
modules/45_carbonprice/NDC/postsolve.gms:75-      + p45_taxCO2eqGlobal2030        * p45_taxCO2eqYearlyIncrease**(t.val-2030)                  * (min(t.val,p45_taxCO2eqConvergenceYear) - p45_lastNDCyear(regi))

modules/29_CES_parameters/calibrate/datainput.gms:70-    
modules/29_CES_parameters/calibrate/datainput.gms:71:  !! in case one input is ppfen/FE and the other Kap
modules/29_CES_parameters/calibrate/datainput.gms:72-  if (sm_tmp eq 1 AND sm_tmp2 eq 1,
--
modules/29_CES_parameters/calibrate/datainput.gms:306-$ifthen.testOneRegi "%optimization%" == "testOneRegi"   !! optimization
modules/29_CES_parameters/calibrate/datainput.gms:307:  !! carry along CES parameters for other regions in testOneRegi runs
modules/29_CES_parameters/calibrate/datainput.gms:308-  pm_cesdata(t,regi,in,cesParameter)$( NOT regi_dyn29(regi) )

modules/45_carbonprice/NDC/preloop.gms:17-        p45_taxCO2eqLastNDCyear(regi) * p45_taxCO2eqYearlyIncrease**(t.val-p45_lastNDCyear(regi)) * (max(p45_taxCO2eqConvergenceYear,t.val) - t.val)
modules/45_carbonprice/NDC/preloop.gms:18:        !! global, weight going from 0 in NDC target year to 1 in and after 2100
modules/45_carbonprice/NDC/preloop.gms:19-      + p45_taxCO2eqGlobal2030          * p45_taxCO2eqYearlyIncrease**(t.val-2030)                    * (min(p45_taxCO2eqConvergenceYear,t.val) - p45_lastNDCyear(regi))

modules/29_CES_parameters/calibrate/bounds.gms:46-      ( pm_cesdata(t,regi,in,"quantity")
modules/29_CES_parameters/calibrate/bounds.gms:47:        !! goes from 0.95 to 1e-12 in -0.2 steps
modules/29_CES_parameters/calibrate/bounds.gms:48-      * max(1e-12, 0.95 + min(0, (1 - sm_CES_calibration_iteration) / sm_tmp))
--
modules/29_CES_parameters/calibrate/bounds.gms:54-  = ( pm_cesdata(t,regi,in,"quantity")
modules/29_CES_parameters/calibrate/bounds.gms:55:    !! goes from 1.05 to 2.05 in +0.2 steps, then jumps to inf
modules/29_CES_parameters/calibrate/bounds.gms:56-    * (1.05 + max(0, (sm_CES_calibration_iteration - 1) / sm_tmp))

modules/29_CES_parameters/calibrate/declarations.gms:90-if (sm_CES_calibration_iteration eq 1,
modules/29_CES_parameters/calibrate/declarations.gms:91:  !! print a comment header giving the order of production factors in the CES
modules/29_CES_parameters/calibrate/declarations.gms:92:  !! tree so that they can be displayed in a meaningful order in calibration
modules/29_CES_parameters/calibrate/declarations.gms:93:  !! reports
modules/29_CES_parameters/calibrate/declarations.gms:94-  file_CES_calibration.pc =  0;   !! text file
--
modules/29_CES_parameters/calibrate/declarations.gms:104-
modules/29_CES_parameters/calibrate/declarations.gms:105:  !! as long as there are nodes pending
modules/29_CES_parameters/calibrate/declarations.gms:106-  while (0 lt card(CES_tp),
modules/29_CES_parameters/calibrate/declarations.gms:107:    !! if the current node is a calibration target and all of its child nodes
modules/29_CES_parameters/calibrate/declarations.gms:108:    !! are pending (so it has not been processed before)
modules/29_CES_parameters/calibrate/declarations.gms:109-    if (sum(CES_tc$(    (   ppf(CES_tc)
--
modules/29_CES_parameters/calibrate/declarations.gms:113-		  ), 1),
modules/29_CES_parameters/calibrate/declarations.gms:114:    !! add the current node to the list
modules/29_CES_parameters/calibrate/declarations.gms:115-    loop (CES_tc, put ", ", CES_tc.tl:0);
--
modules/29_CES_parameters/calibrate/declarations.gms:117-
modules/29_CES_parameters/calibrate/declarations.gms:118:  !! if no child nodes are pending
modules/29_CES_parameters/calibrate/declarations.gms:119-  if (0 eq sum(cesOut2cesIn(CES_tc,CES_tp), 1),
modules/29_CES_parameters/calibrate/declarations.gms:120:    !! remove current node from pending nodes
modules/29_CES_parameters/calibrate/declarations.gms:121-    CES_tp(CES_tc) = NO;
modules/29_CES_parameters/calibrate/declarations.gms:122:    !! set parent node as current node
modules/29_CES_parameters/calibrate/declarations.gms:123-    loop (cesOut2cesIn(out,CES_tc), CES_ts(out) = YES);
--
modules/29_CES_parameters/calibrate/declarations.gms:127-
modules/29_CES_parameters/calibrate/declarations.gms:128:  !! if any child nodes are pending
modules/29_CES_parameters/calibrate/declarations.gms:129-  else
modules/29_CES_parameters/calibrate/declarations.gms:130:    !! set first pending child node as current node
modules/29_CES_parameters/calibrate/declarations.gms:131-    loop (cesOut2cesIn(CES_tc,CES_tp), CES_ts(CES_tp) = YES);
--
modules/29_CES_parameters/calibrate/declarations.gms:140-
modules/29_CES_parameters/calibrate/declarations.gms:141:  !! print the csv header
modules/29_CES_parameters/calibrate/declarations.gms:142-  file_CES_calibration.pc =  5;   !! csv file

modules/11_aerosols/exoGAINS/datainput.gms:128-loop ((t,regi)$( t.val ge 2005 ),
modules/11_aerosols/exoGAINS/datainput.gms:129:  !! share in solids
modules/11_aerosols/exoGAINS/datainput.gms:130-  if (sum(fe2ppfen("fesos",in), p11_cesIO(t,regi,in)) gt 0,
--
modules/11_aerosols/exoGAINS/datainput.gms:146-    else
modules/11_aerosols/exoGAINS/datainput.gms:147:      !! When calibrating to a new region set with insufficient data coverage in
modules/11_aerosols/exoGAINS/datainput.gms:148:      !! the gdx, vm_cesIO will be all zero.  In that case, simply split 50/50.
modules/11_aerosols/exoGAINS/datainput.gms:149-      p11_share_sector(t,"pebiolc","sesobio","biotr","indst",regi) = 0.5;
--
modules/11_aerosols/exoGAINS/datainput.gms:158-
modules/11_aerosols/exoGAINS/datainput.gms:159:  !! share in liquids
modules/11_aerosols/exoGAINS/datainput.gms:160-  if (sum(fe2ppfen("fehos",in), p11_cesIO(t,regi,in)) gt 0,
--
modules/11_aerosols/exoGAINS/datainput.gms:182-
modules/11_aerosols/exoGAINS/datainput.gms:183:  !! share in gases
modules/11_aerosols/exoGAINS/datainput.gms:184-  if (sum(fe2ppfen("fegas",in), p11_cesIO(t,regi,in)) gt 0,

modules/46_carbonpriceRegi/NDC/postsolve.gms:58-  max(
modules/46_carbonpriceRegi/NDC/postsolve.gms:59:      !! set lower bound of 0.1 $/t to avoid that the model never gets the carbon price back up
modules/46_carbonpriceRegi/NDC/postsolve.gms:60-      0.1 * sm_DptCO2_2_TDpGtC,
modules/46_carbonpriceRegi/NDC/postsolve.gms:61:      !! set regi tax such that total CO2 tax changes by desired factor, taking into account changed pm_taxCO2eq taxes
modules/46_carbonpriceRegi/NDC/postsolve.gms:62-      p46_factorRescaleCO2TaxLtd(t,regi) * (pm_taxCO2eqRegi(t,regi) + p46_taxCO2eqLast(t,regi))
--
modules/46_carbonpriceRegi/NDC/postsolve.gms:85-        p46_taxCO2eqLastNDCyear(regi) * p46_taxCO2eqYearlyIncrease**(t.val-p46_lastNDCyear(regi)) * (max(p46_taxCO2eqConvergenceYear,t.val) - t.val)
modules/46_carbonpriceRegi/NDC/postsolve.gms:86:        !! global, weight going from 0 in NDC target year to 1 in and after 2100
modules/46_carbonpriceRegi/NDC/postsolve.gms:87-      + p46_taxCO2eqGlobal2030          * p46_taxCO2eqYearlyIncrease**(t.val-2030)                    * (min(t.val,p46_taxCO2eqConvergenceYear) - p46_lastNDCyear(regi))

modules/46_carbonpriceRegi/NDC/preloop.gms:22-        p46_taxCO2eqLastNDCyear(regi) * p46_taxCO2eqYearlyIncrease**(t.val-p46_lastNDCyear(regi)) * (max(p46_taxCO2eqConvergenceYear,t.val) - t.val)
modules/46_carbonpriceRegi/NDC/preloop.gms:23:        !! global, weight going from 0 in NDC target year to 1 in and after 2100
modules/46_carbonpriceRegi/NDC/preloop.gms:24-      + p46_taxCO2eqGlobal2030          * p46_taxCO2eqYearlyIncrease**(t.val-2030)                * (min(p46_taxCO2eqConvergenceYear,t.val) - p46_lastNDCyear(regi))

modules/29_CES_parameters/calibrate/preloop.gms:51-if( sm_CES_calibration_iteration eq 1 and s29_CES_calibration_new_structure eq 1,
modules/29_CES_parameters/calibrate/preloop.gms:52:  !! Set CES prices to the value specified by cm_CES_calibration_default_prices
modules/29_CES_parameters/calibrate/preloop.gms:53:  !! and abort if cm_CES_calibration_default_prices == 0
modules/29_CES_parameters/calibrate/preloop.gms:54-$ifthen.default_prices %cm_CES_calibration_default_prices% == "0"
--
modules/29_CES_parameters/calibrate/preloop.gms:66-else
modules/29_CES_parameters/calibrate/preloop.gms:67:  !! If not first iteration or known CES structure, compute ppf prices from CES derivatives load from file
modules/29_CES_parameters/calibrate/preloop.gms:68-
modules/29_CES_parameters/calibrate/preloop.gms:69:  !! Compute prices of each node from CES derivatives of previous run:
modules/29_CES_parameters/calibrate/preloop.gms:70:  !! d(V_o)/d(V_i) = pi_i  = xi_i * eff_i * effGr_i * V_o**(1-rho_o)  *  (eff_i * effGr_i * V_i)**(rho_o-1)
modules/29_CES_parameters/calibrate/preloop.gms:71-  p29_CESderivative(t,regi_dyn29(regi),ces_29(out,in))$(
--
modules/29_CES_parameters/calibrate/preloop.gms:90-
modules/29_CES_parameters/calibrate/preloop.gms:91:  !! Propagate price down the CES tree to get prices in terms of inco,
modules/29_CES_parameters/calibrate/preloop.gms:92:  !! i.e. calc d(inco)/d(in) by applying the chain rule (product of node derivatives)
modules/29_CES_parameters/calibrate/preloop.gms:93:  !! Going down, iteratively compute each income derivative as product of all prices in branch above
modules/29_CES_parameters/calibrate/preloop.gms:94:  !! Ppf prices will later be used as: pm_cesdata(in,"price") = p29_CESderivative("inco",in).
modules/29_CES_parameters/calibrate/preloop.gms:95:  !! the rest is discarded.
modules/29_CES_parameters/calibrate/preloop.gms:96-  loop ((cesLevel2cesIO(counter,in),ces_29(in,in2),ces2_29(in2,in3)),
modules/29_CES_parameters/calibrate/preloop.gms:97:      !! in3 is the current node, in2 is the node above.
modules/29_CES_parameters/calibrate/preloop.gms:98-      p29_CESderivative(t,regi_dyn29(regi),"inco",in3) !!   d(inco)/d(in3)
--
modules/29_CES_parameters/calibrate/preloop.gms:103-
modules/29_CES_parameters/calibrate/preloop.gms:104:  !! Prices of intermediate production factors are all set to 1,
modules/29_CES_parameters/calibrate/preloop.gms:105:  !! To account for the chain rule multiplication above.
modules/29_CES_parameters/calibrate/preloop.gms:106:  !! except on the level above the perfect substitutes if they are ppf_29.
modules/29_CES_parameters/calibrate/preloop.gms:107:  !! Here, the level above gets the price, while the perfect substitutes below get 1.
modules/29_CES_parameters/calibrate/preloop.gms:108-  loop ( cesOut2cesIn(in2,in) $ (
--
modules/29_CES_parameters/calibrate/preloop.gms:115-
modules/29_CES_parameters/calibrate/preloop.gms:116:  !! Prices of perfect substitutes factors are all 1
modules/29_CES_parameters/calibrate/preloop.gms:117-  p29_CESderivative(t,regi_dyn29(regi),out,ppf_29(in2))$(
--
modules/29_CES_parameters/calibrate/preloop.gms:119-    = 1;
modules/29_CES_parameters/calibrate/preloop.gms:120:  !! Price of inco is 1, too
modules/29_CES_parameters/calibrate/preloop.gms:121-  p29_cesdata_load(t,regi_dyn29(regi),"inco","price") = 1;   !! unit price
modules/29_CES_parameters/calibrate/preloop.gms:122-
modules/29_CES_parameters/calibrate/preloop.gms:123:  !! Transfer prices
modules/29_CES_parameters/calibrate/preloop.gms:124-  pm_cesdata(t,regi_dyn29(regi), in, "price") =
--
modules/29_CES_parameters/calibrate/preloop.gms:131-
modules/29_CES_parameters/calibrate/preloop.gms:132:  !! The calibration of elasticities of substitution takes
modules/29_CES_parameters/calibrate/preloop.gms:133:  !! much longer to converge if it starts from a high elasticity of
modules/29_CES_parameters/calibrate/preloop.gms:134:  !! substitution. To avoid this situation, the price of the capital stock
modules/29_CES_parameters/calibrate/preloop.gms:135:  !! is increased
modules/29_CES_parameters/calibrate/preloop.gms:136-  if (sm_CES_calibration_iteration eq 1,
--
modules/29_CES_parameters/calibrate/preloop.gms:143-
modules/29_CES_parameters/calibrate/preloop.gms:144:  !! Write prices to file and abort, to use them in calibration with differing
modules/29_CES_parameters/calibrate/preloop.gms:145:  !! CES structure
modules/29_CES_parameters/calibrate/preloop.gms:146-$ifthen.write_prices %c_CES_calibration_write_prices% == "1"
--
modules/29_CES_parameters/calibrate/preloop.gms:484-
modules/29_CES_parameters/calibrate/preloop.gms:485:!! Write to file
modules/29_CES_parameters/calibrate/preloop.gms:486-if (sm_CES_calibration_iteration eq 1, !! first CES calibration iteration
--
modules/29_CES_parameters/calibrate/preloop.gms:539-
modules/29_CES_parameters/calibrate/preloop.gms:540:  !! compute the total for factors that are ppf in the CES and ipf in the
modules/29_CES_parameters/calibrate/preloop.gms:541:  !! putty. For the first period, we assume that pm_cesdata stays constant
modules/29_CES_parameters/calibrate/preloop.gms:542-  pm_cesdata(t,regi_dyn29,out,"quantity")$( t0(t) AND ppfIO_putty(out) )
--
modules/29_CES_parameters/calibrate/preloop.gms:545-
modules/29_CES_parameters/calibrate/preloop.gms:546:  !! compute the total for factors that are ppf in the CES and ipf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:547-  pm_cesdata(t,regi_dyn29,out,"quantity")$(
--
modules/29_CES_parameters/calibrate/preloop.gms:598-
modules/29_CES_parameters/calibrate/preloop.gms:599:!! If there has been a rescaling for historical steps, repeat previous steps with new prices
modules/29_CES_parameters/calibrate/preloop.gms:600-if ( sm_tmp2 gt 0, !! If there has been a rescaling
--
modules/29_CES_parameters/calibrate/preloop.gms:658-
modules/29_CES_parameters/calibrate/preloop.gms:659:    !! compute the total for factors that are ppf in the CES and ipf in the
modules/29_CES_parameters/calibrate/preloop.gms:660:    !! putty. For the first period, we assume that pm_cesdata stays constant
modules/29_CES_parameters/calibrate/preloop.gms:661-    pm_cesdata(t,regi_dyn29,out,"quantity")$( t0(t) AND ppfIO_putty(out) )
--
modules/29_CES_parameters/calibrate/preloop.gms:664-
modules/29_CES_parameters/calibrate/preloop.gms:665:    !! compute the total for factors that are ppf in the CES and ipf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:666-    pm_cesdata(t,regi_dyn29,out,"quantity")$(
--
modules/29_CES_parameters/calibrate/preloop.gms:801-
modules/29_CES_parameters/calibrate/preloop.gms:802:!! do either en or kap exceed inco?
modules/29_CES_parameters/calibrate/preloop.gms:803-loop (cesOut2cesIn("inco",in)$( NOT sameas(in,"lab") ),
--
modules/29_CES_parameters/calibrate/preloop.gms:918-
modules/29_CES_parameters/calibrate/preloop.gms:919:  !! if prices haven't already been loaded
modules/29_CES_parameters/calibrate/preloop.gms:920-  if (sm_CES_calibration_iteration > 1 or s29_CES_calibration_new_structure eq 0,
modules/29_CES_parameters/calibrate/preloop.gms:921-
modules/29_CES_parameters/calibrate/preloop.gms:922:    !! Compute ppf prices from CES derivatives of previous run
modules/29_CES_parameters/calibrate/preloop.gms:923-    p29_CESderivative(t,regi_dyn29(regi),cesOut2cesIn(out,in))$(
--
modules/29_CES_parameters/calibrate/preloop.gms:943-
modules/29_CES_parameters/calibrate/preloop.gms:944:    !! Propagate price down the CES tree
modules/29_CES_parameters/calibrate/preloop.gms:945-    loop ((cesLevel2cesIO(counter,in),cesOut2cesIn(in,in2),cesOut2cesIn2(in2,in3)),
--
modules/29_CES_parameters/calibrate/preloop.gms:951-
modules/29_CES_parameters/calibrate/preloop.gms:952:    !! Prices of intermediate production factors are all 1, except on the level
modules/29_CES_parameters/calibrate/preloop.gms:953:    !! above the perfect substitutes if they are ppf_29
modules/29_CES_parameters/calibrate/preloop.gms:954-    loop (cesOut2cesIn(in2,in)$(
--
modules/29_CES_parameters/calibrate/preloop.gms:960-
modules/29_CES_parameters/calibrate/preloop.gms:961:    !!  Prices of perfect substitutes factors are all 1
modules/29_CES_parameters/calibrate/preloop.gms:962-    p29_CESderivative(t,regi_dyn29(regi),out,ppf_beyondcalib_29(in2))$(
--
modules/29_CES_parameters/calibrate/preloop.gms:975-$ifthen.FE_target "%c_CES_calibration_industry_FE_target%" == "1" !! c_CES_calibration_industry_FE_target
modules/29_CES_parameters/calibrate/preloop.gms:976:    !! set minimum price on ppf_industry
modules/29_CES_parameters/calibrate/preloop.gms:977-    pm_cesdata(t,regi_dyn29(regi),ppf_industry_dyn37(in),"price")
--
modules/29_CES_parameters/calibrate/preloop.gms:981-
modules/29_CES_parameters/calibrate/preloop.gms:982:    !! smooth historical prices
modules/29_CES_parameters/calibrate/preloop.gms:983-    pm_cesdata(t_29hist(t),regi_dyn29(regi),in,"price")$(
--
modules/29_CES_parameters/calibrate/preloop.gms:991-  else
modules/29_CES_parameters/calibrate/preloop.gms:992:    !! complements are not treated in the first iteration
modules/29_CES_parameters/calibrate/preloop.gms:993-    pm_cesdata(t,regi,ipf_beyond_29(in),"price")$( NOT ue_industry_dyn37(in) )
--
modules/29_CES_parameters/calibrate/preloop.gms:997-
modules/29_CES_parameters/calibrate/preloop.gms:998:  !! The calibration of elasticities of substitution takes much longer to
modules/29_CES_parameters/calibrate/preloop.gms:999:  !! converge if it starts from a high elasticity of substitution. To avoid
modules/29_CES_parameters/calibrate/preloop.gms:1000:  !! this situation, the price of the capital stock is increased.
modules/29_CES_parameters/calibrate/preloop.gms:1001-  if (sm_CES_calibration_iteration eq 1,
--
modules/29_CES_parameters/calibrate/preloop.gms:1009-
modules/29_CES_parameters/calibrate/preloop.gms:1010:  !! Report prices based on the inco marginal, before they are scaled.
modules/29_CES_parameters/calibrate/preloop.gms:1011-  put capital_unit;
--
modules/29_CES_parameters/calibrate/preloop.gms:1031-
modules/29_CES_parameters/calibrate/preloop.gms:1032:  !! First, we compute the quantity for the root deriving from the ppf
modules/29_CES_parameters/calibrate/preloop.gms:1033:  !! quantities and prices and we adjust the ppf prices so that it matches the
modules/29_CES_parameters/calibrate/preloop.gms:1034:  !! root quantity. The current formulation does not support putty in beyond
modules/29_CES_parameters/calibrate/preloop.gms:1035:  !! and complements.
modules/29_CES_parameters/calibrate/preloop.gms:1036-
--
modules/29_CES_parameters/calibrate/preloop.gms:1076-    loop (cesOut2cesIn_below(out,ppf(in)),
modules/29_CES_parameters/calibrate/preloop.gms:1077:     !! adjust the price for historical periods
modules/29_CES_parameters/calibrate/preloop.gms:1078-      pm_cesdata(t,regi,in,"price") $ (
--
modules/29_CES_parameters/calibrate/preloop.gms:1081-      / sm_tmp;
modules/29_CES_parameters/calibrate/preloop.gms:1082:     !! adjust the quantity for scenario periods
modules/29_CES_parameters/calibrate/preloop.gms:1083-      pm_cesdata(t,regi,in,"quantity") $ (
--
modules/29_CES_parameters/calibrate/preloop.gms:1103-
modules/29_CES_parameters/calibrate/preloop.gms:1104:    !! compute the total for factors that are ppf in the CES and ipf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:1105-    pm_cesdata(t0(t_29),regi_dyn29,ppfIO_putty(out),"quantity")
--
modules/29_CES_parameters/calibrate/preloop.gms:1108-
modules/29_CES_parameters/calibrate/preloop.gms:1109:    !! compute the delta for factors that are ipf in the CES and ppf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:1110-    pm_cesdata_putty(t0(t_29),regi_dyn29,ppf_putty(out),"quantity")
--
modules/29_CES_parameters/calibrate/preloop.gms:1115-
modules/29_CES_parameters/calibrate/preloop.gms:1116:    !! compute the total for factors that are ppf in the CES and ipf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:1117-    pm_cesdata(t_29,regi_dyn29,ppfIO_putty(out),"quantity")$( NOT t0(t_29) )
--
modules/29_CES_parameters/calibrate/preloop.gms:1128-
modules/29_CES_parameters/calibrate/preloop.gms:1129:    !! compute the delta for factors that are ipf in the CES and ppf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:1130-    pm_cesdata_putty(t_29,regi_dyn29,ppf_putty(out),"quantity")$( NOT t0(t_29) )
--
modules/29_CES_parameters/calibrate/preloop.gms:1180-                                      AND pm_cesdata_sigma(ttot,out) ne -1 )
modules/29_CES_parameters/calibrate/preloop.gms:1181:    !! Do not compute it if sigma = 0, because these should be estimated
modules/29_CES_parameters/calibrate/preloop.gms:1182-  = 1 - (1 / pm_cesdata_sigma(ttot,out));
--
modules/29_CES_parameters/calibrate/preloop.gms:1439-
modules/29_CES_parameters/calibrate/preloop.gms:1440:!! - adjust efficiency parameters for feelhth_X and feh2_X
modules/29_CES_parameters/calibrate/preloop.gms:1441-$ifthen.industry_FE_target "%c_CES_calibration_industry_FE_target%" == "0"
--
modules/29_CES_parameters/calibrate/preloop.gms:1445-                          AND NOT cesOut2cesIn_below("ue_steel_secondary",in) ),
modules/29_CES_parameters/calibrate/preloop.gms:1446:  !! in2 is the reference energy input (gas if 'in' is H2)
modules/29_CES_parameters/calibrate/preloop.gms:1447-  loop (in2$( pm_calibrate_eff_scale(in,in2,"level") ),
modules/29_CES_parameters/calibrate/preloop.gms:1448:    !! compute the parameter describing the speed of convergence towards in2
modules/29_CES_parameters/calibrate/preloop.gms:1449-    p29_t_tmp(t)$( t_29scen(t) )
--
modules/29_CES_parameters/calibrate/preloop.gms:1614-
modules/29_CES_parameters/calibrate/preloop.gms:1615:!! compute the total for factors that are ppf in the CES and ipf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:1616-p29_test_CES_recursive(t0(t),regi,ppfIO_putty(out))
--
modules/29_CES_parameters/calibrate/preloop.gms:1619-
modules/29_CES_parameters/calibrate/preloop.gms:1620:!! compute the total for factors that are ppf in the CES and ipf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:1621-p29_test_CES_recursive(t,regi_dyn29,ppfIO_putty(out))$( NOT t0(t) )
--
modules/29_CES_parameters/calibrate/preloop.gms:1632-
modules/29_CES_parameters/calibrate/preloop.gms:1633:!! compute the delta for factors that are ipf in the CES and ppf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:1634-p29_test_CES_putty_recursive(t,regi_dyn29,ppf_putty(out))$( NOT t0(t) )
--
modules/29_CES_parameters/calibrate/preloop.gms:1665-  p29_test_CES_recursive(t,regi,out)$( NOT ipf_putty(out) )
modules/29_CES_parameters/calibrate/preloop.gms:1666:  !! use exp(log(a) * b) = a ** b because the latter is not accurate in GAMS
modules/29_CES_parameters/calibrate/preloop.gms:1667-  = exp(
--
modules/29_CES_parameters/calibrate/preloop.gms:1684-
modules/29_CES_parameters/calibrate/preloop.gms:1685:  !! compute the total for factors that are ppf in the CES and ipf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:1686-  p29_test_CES_recursive(t,regi_dyn29,out)$( ppfIO_putty(out) )
--
modules/29_CES_parameters/calibrate/preloop.gms:1697-
modules/29_CES_parameters/calibrate/preloop.gms:1698:  !! compute the delta for factors that are ipf in the CES and ppf in the putty
modules/29_CES_parameters/calibrate/preloop.gms:1699-  p29_test_CES_putty_recursive(t,regi_dyn29,out)$( ppf_putty(out) )

modules/11_aerosols/exoGAINS/presolve.gms:9-loop ((t,regi)$( t.val ge 2005 ),
modules/11_aerosols/exoGAINS/presolve.gms:10:  !! share in solids
modules/11_aerosols/exoGAINS/presolve.gms:11-  if (sum(fe2ppfen("fesos",in), p11_cesIO(t,regi,in)) gt 0,
--
modules/11_aerosols/exoGAINS/presolve.gms:27-    else
modules/11_aerosols/exoGAINS/presolve.gms:28:      !! When calibrating to a new region set with insufficient data coverage in
modules/11_aerosols/exoGAINS/presolve.gms:29:      !! the gdx, vm_cesIO will be all zero.  In that case, simply split 50/50.
modules/11_aerosols/exoGAINS/presolve.gms:30-      p11_share_sector(t,"pebiolc","sesobio","biotr","indst",regi) = 0.5;
--
modules/11_aerosols/exoGAINS/presolve.gms:39-
modules/11_aerosols/exoGAINS/presolve.gms:40:  !! share in liquids
modules/11_aerosols/exoGAINS/presolve.gms:41-  if (sum(fe2ppfen("fehos",in), p11_cesIO(t,regi,in)) gt 0,
--
modules/11_aerosols/exoGAINS/presolve.gms:63-
modules/11_aerosols/exoGAINS/presolve.gms:64:  !! share in gases
modules/11_aerosols/exoGAINS/presolve.gms:65-  if (sum(fe2ppfen("fegas",in), p11_cesIO(t,regi,in)) gt 0,

modules/36_buildings/services_putty/presolve.gms:126-      / pm_fe2es(t,regi,teEs)
modules/36_buildings/services_putty/presolve.gms:127:      !! add taxes, subsidies, and later on costs
modules/36_buildings/services_putty/presolve.gms:128-      ;
--
modules/36_buildings/services_putty/presolve.gms:194-    
modules/36_buildings/services_putty/presolve.gms:195:    !! give a high parameter value to the technologies that do not have some
modules/36_buildings/services_putty/presolve.gms:196-   p36_logitCalibration(t2,regi_dyn36(regi),entyFe,esty,teEs) $ (fe2es_dyn36(entyFe,esty,teEs)
--
modules/36_buildings/services_putty/presolve.gms:202-     
modules/36_buildings/services_putty/presolve.gms:203:     !! Decrease the calibration factor for some technologies, based on the difference between the 2015 (ttot) Income per capita and scenario (t2) income per capita.
modules/36_buildings/services_putty/presolve.gms:204:     !! the calibration factor decreases by 90% when income reaches 30 k$ if income per capita was equal or below 5 k$ in 2015.
modules/36_buildings/services_putty/presolve.gms:205:     !! the decrease is lower if the starting income was above 10k$ and is 0 if income was above 30k$
modules/36_buildings/services_putty/presolve.gms:206-   p36_logitCalibration(t2,regi_dyn36(regi),entyFe,esty,teEs) $ richTechs(teEs) =  !! calib = calib * (1 - 0.90 * X) 

modules/36_buildings/services_with_capital/presolve.gms:127-      / pm_fe2es(t,regi,teEs)
modules/36_buildings/services_with_capital/presolve.gms:128:      !! add taxes, subsidies, and later on costs
modules/36_buildings/services_with_capital/presolve.gms:129-      ;
--
modules/36_buildings/services_with_capital/presolve.gms:195-    
modules/36_buildings/services_with_capital/presolve.gms:196:    !! give a high parameter value to the technologies that do not have some
modules/36_buildings/services_with_capital/presolve.gms:197-   p36_logitCalibration(t2,regi_dyn36(regi),entyFe,esty,teEs) $ (fe2es_dyn36(entyFe,esty,teEs)
--
modules/36_buildings/services_with_capital/presolve.gms:203-     
modules/36_buildings/services_with_capital/presolve.gms:204:     !! Decrease the calibration factor for some technologies, based on the difference between the 2015 (ttot) Income per capita and scenario (t2) income per capita.
modules/36_buildings/services_with_capital/presolve.gms:205:     !! the calibration factor decreases by 90% when income reaches 30 k$ if income per capita was equal or below 5 k$ in 2015.
modules/36_buildings/services_with_capital/presolve.gms:206:     !! the decrease is lower if the starting income was above 10k$ and is 0 if income was above 30k$
modules/36_buildings/services_with_capital/presolve.gms:207-   p36_logitCalibration(t2,regi_dyn36(regi),entyFe,esty,teEs) $ richTechs(teEs) =  !! calib = calib * (1 - 0.90 * X) 

modules/37_industry/subsectors/sets.gms:252-
modules/37_industry/subsectors/sets.gms:253:  !! Calibration Sets
modules/37_industry/subsectors/sets.gms:254-  pf_eff_target_dyn37(all_in)    "production factors with efficiency target"

modules/37_industry/subsectors/postsolve.gms:50-    * sum(se2fe(entySE2,entyFE,te),
modules/37_industry/subsectors/postsolve.gms:51:        !! collapse entySE dimension, so emission factors apply to all entyFE
modules/37_industry/subsectors/postsolve.gms:52-	!! regardless or origin, and therefore entySEbio and entySEsyn have

modules/37_industry/subsectors/datainput.gms:126-  * 1e9;                   !! * t/Gt
modules/37_industry/subsectors/datainput.gms:127:                           !! = TWa/Gt
modules/37_industry/subsectors/datainput.gms:128-
--
modules/37_industry/subsectors/datainput.gms:140-sm_tmp2 = 0.75;   !! maximum "efficiency gain", from 2015 baseline value to 
modules/37_industry/subsectors/datainput.gms:141:                  !! thermodynamic limit
modules/37_industry/subsectors/datainput.gms:142-sm_tmp  = 2050;   !! period in which closing could be achieved
--
modules/37_industry/subsectors/datainput.gms:154-
modules/37_industry/subsectors/datainput.gms:155:  !! To account for strong 2015-20 drops due to imperfect 2020 energy data,
modules/37_industry/subsectors/datainput.gms:156:  !! use the lower of the calculated curve, or 95 % of the baseline specific
modules/37_industry/subsectors/datainput.gms:157:  !! energy demand
modules/37_industry/subsectors/datainput.gms:158-  p37_energy_limit_slope(ttot,regi,out)$( ttot.val ge 2015 )
--
modules/37_industry/subsectors/datainput.gms:181-
modules/37_industry/subsectors/datainput.gms:182:  !! To account for strong 2015-20 drops due to imperfect 2020 energy data,
modules/37_industry/subsectors/datainput.gms:183:  !! use the lower of the calculated curve, or 95 % of the baseline specific
modules/37_industry/subsectors/datainput.gms:184:  !! energy demand
modules/37_industry/subsectors/datainput.gms:185-  p37_energy_limit_slope(ttot,regi,out)$( ttot.val ge 2015 )
--
modules/37_industry/subsectors/datainput.gms:294-    p37_industry_quantity_targets(ttot,all_regi,all_in)   "quantity targets for industry in policy scenarios"
modules/37_industry/subsectors/datainput.gms:295:    !! from FORECAST v1.0_8Gt_Bal.xlsx
modules/37_industry/subsectors/datainput.gms:296-    /
--
modules/37_industry/subsectors/datainput.gms:322-
modules/37_industry/subsectors/datainput.gms:323:  !! convert Mt to Gt
modules/37_industry/subsectors/datainput.gms:324-  p37_industry_quantity_targets(t,regi,in)$(
--
modules/37_industry/subsectors/datainput.gms:326-    = p37_industry_quantity_targets(t,regi,in)
modules/37_industry/subsectors/datainput.gms:327:      !! Mt/yr * 1e-3 Gt/Mt = Gt/yr
modules/37_industry/subsectors/datainput.gms:328-    * 1e-3;
modules/37_industry/subsectors/datainput.gms:329-
modules/37_industry/subsectors/datainput.gms:330:  !! extend beyond 2050
modules/37_industry/subsectors/datainput.gms:331:  !! FIXME: do this smarter, using something like GDPpC growth or something
modules/37_industry/subsectors/datainput.gms:332-  p37_industry_quantity_targets(t,regi,in)$(
--
modules/37_industry/subsectors/datainput.gms:340-    p37_industry_quantity_targets(ttot,all_regi,all_in)   "quantity targets for industry in policy scenarios"
modules/37_industry/subsectors/datainput.gms:341:    !! from Ariadne_Industrieproduktion_Harmonisierung.xlsx
modules/37_industry/subsectors/datainput.gms:342-    /
--
modules/37_industry/subsectors/datainput.gms:365-
modules/37_industry/subsectors/datainput.gms:366:  !! convert Mt to Gt
modules/37_industry/subsectors/datainput.gms:367-  p37_industry_quantity_targets(t,regi,in)$(
--
modules/37_industry/subsectors/datainput.gms:369-    = p37_industry_quantity_targets(t,regi,in)
modules/37_industry/subsectors/datainput.gms:370:      !! Mt/yr * 1e-3 Gt/Mt = Gt/yr
modules/37_industry/subsectors/datainput.gms:371-    * 1e-3;
modules/37_industry/subsectors/datainput.gms:372-
modules/37_industry/subsectors/datainput.gms:373:  !! extend beyond 2045
modules/37_industry/subsectors/datainput.gms:374:  !! FIXME: do this smarter, using something like GDPpC growth or something
modules/37_industry/subsectors/datainput.gms:375-  p37_industry_quantity_targets(t,regi,in)$(

modules/37_industry/subsectors/bounds.gms:24-$ifthen.rcp_scen "%cm_rcp_scen%" == "none"
modules/37_industry/subsectors/bounds.gms:25:  !! In no-policy scenarios, tight bounds representing usual scrap recycling
modules/37_industry/subsectors/bounds.gms:26:  !! rates apply.  Only 10% of the difference between projected secondary
modules/37_industry/subsectors/bounds.gms:27:  !! steel production and the upper bound with increased recycling rates are
modules/37_industry/subsectors/bounds.gms:28:  !! available for increased production.
modules/37_industry/subsectors/bounds.gms:29-  vm_cesIO.up(t,regi,"ue_steel_secondary")
--
modules/37_industry/subsectors/bounds.gms:38-$elseif.rcp_scen "%cm_rcp_scen%" == "rcp85"
modules/37_industry/subsectors/bounds.gms:39:  !! In no-policy scenarios, tight bounds representing usual scrap recycling
modules/37_industry/subsectors/bounds.gms:40:  !! rates apply.  Only 10% of the difference between projected secondary
modules/37_industry/subsectors/bounds.gms:41:  !! steel production and the upper bound with increased recycling rates are
modules/37_industry/subsectors/bounds.gms:42:  !! available for increased production.
modules/37_industry/subsectors/bounds.gms:43-  vm_cesIO.up(t,regi,"ue_steel_secondary")
--
modules/37_industry/subsectors/bounds.gms:52-$elseif.rcp_scen "%cm_rcp_scen%" == "rcp60"
modules/37_industry/subsectors/bounds.gms:53:  !! In no-policy scenarios, tight bounds representing usual scrap recycling
modules/37_industry/subsectors/bounds.gms:54:  !! rates apply.  Only 10% of the difference between projected secondary
modules/37_industry/subsectors/bounds.gms:55:  !! steel production and the upper bound with increased recycling rates are
modules/37_industry/subsectors/bounds.gms:56:  !! available for increased production.
modules/37_industry/subsectors/bounds.gms:57-  vm_cesIO.up(t,regi,"ue_steel_secondary")
--
modules/37_industry/subsectors/bounds.gms:66-$elseif.rcp_scen "%cm_rcp_scen%" == "rcp45"
modules/37_industry/subsectors/bounds.gms:67:  !! In no-policy scenarios, tight bounds representing usual scrap recycling
modules/37_industry/subsectors/bounds.gms:68:  !! rates apply.  Only 10% of the difference between projected secondary
modules/37_industry/subsectors/bounds.gms:69:  !! steel production and the upper bound with increased recycling rates are
modules/37_industry/subsectors/bounds.gms:70:  !! available for increased production.
modules/37_industry/subsectors/bounds.gms:71-  vm_cesIO.up(t,regi,"ue_steel_secondary")
--
modules/37_industry/subsectors/bounds.gms:80-$else.rcp_scen
modules/37_industry/subsectors/bounds.gms:81:  !! In policy scenarios, secondary steel production can be increased up to the
modules/37_industry/subsectors/bounds.gms:82:  !! limit of theoretical scrap availability.
modules/37_industry/subsectors/bounds.gms:83-  vm_cesIO.up(t,regi,"ue_steel_secondary")
--
modules/37_industry/subsectors/bounds.gms:126-
modules/37_industry/subsectors/bounds.gms:127:!! Fix industry output for Bal and EnSec scenario
modules/37_industry/subsectors/bounds.gms:128-$if "%cm_indstExogScen%" == "forecast_bal"   $set cm_indstExogScen_set "YES"

modules/37_industry/subsectors/equations.gms:105-$ifthen.fixed_production "%cm_import_EU%" == "bal"   !! cm_import_EU
modules/37_industry/subsectors/equations.gms:106:         !! do not limit steel production shares for fixed production
modules/37_industry/subsectors/equations.gms:107-     AND p37_industry_quantity_targets(ttot,regi,"ue_steel_secondary") eq 0
--
modules/37_industry/subsectors/equations.gms:109-$ifthen.exogDem_scen NOT "%cm_exogDem_scen%" == "off"
modules/37_industry/subsectors/equations.gms:110:         !! do not limit steel production shares for fixed production
modules/37_industry/subsectors/equations.gms:111-     AND pm_exogDemScen(ttot,regi,"%cm_exogDem_scen%","ue_steel_secondary") eq 0
--
modules/37_industry/subsectors/equations.gms:142-  =e=
modules/37_industry/subsectors/equations.gms:143:    !! map sub-sector emissions to sub-sector MACs
modules/37_industry/subsectors/equations.gms:144:    !! otherInd has no CCS, therefore no MAC, cement has both fuel and process
modules/37_industry/subsectors/equations.gms:145:    !! emissions under the same MAC
modules/37_industry/subsectors/equations.gms:146-    sum(emiMac2mac(emiInd37,macInd37),
modules/37_industry/subsectors/equations.gms:147:      !! add cement process emissions, which are calculated in core/preloop
modules/37_industry/subsectors/equations.gms:148:      !! from a econometric fit and might not correspond to energy use (FIXME)
modules/37_industry/subsectors/equations.gms:149-      ( sum((secInd37_2_emiINd37(secInd37,emiInd37),entyFE),

modules/37_industry/fixed_shares/sets.gms:143-
modules/37_industry/fixed_shares/sets.gms:144:  !! empty sets for subsectors compatibility
modules/37_industry/fixed_shares/sets.gms:145-  industry_ue_calibration_target_dyn37(all_in)   ""   / /

modules/37_industry/fixed_shares/presolve.gms:14-
modules/37_industry/fixed_shares/presolve.gms:15:  !! lowest price for which abatement equals current abatement
modules/37_industry/fixed_shares/presolve.gms:16-  pm_CementAbatementPrice(ttot,regi)$( ttot.val ge 2005 )
--
modules/37_industry/fixed_shares/presolve.gms:27-
modules/37_industry/fixed_shares/presolve.gms:28:  !! mix prices of residual and abated emissions
modules/37_industry/fixed_shares/presolve.gms:29-  pm_CementAbatementPrice(ttot,regi)$( ttot.val ge 2005 )
--
modules/37_industry/fixed_shares/presolve.gms:45-  
modules/37_industry/fixed_shares/presolve.gms:46:  !! Demand can only be reduced by 1% p.a.
modules/37_industry/fixed_shares/presolve.gms:47-  loop (ttot$( ttot.val gt 2005 ),

modules/37_industry/fixed_shares/postsolve.gms:33-    * sum(se2fe(entySE2,entyFE,te),
modules/37_industry/fixed_shares/postsolve.gms:34:        !! collapse entySE dimension, so emission factors apply to all entyFE
modules/37_industry/fixed_shares/postsolve.gms:35-	!! regardless or origin, and therefore entySEbio and entySEsyn have

modules/37_industry/fixed_shares/declarations.gms:33-
modules/37_industry/fixed_shares/declarations.gms:34:  !! empty parameters from subsectors
modules/37_industry/fixed_shares/declarations.gms:35-  pm_energy_limit(all_in)    ""   / /

modules/32_power/RLDC/equations.gms:295-;
modules/32_power/RLDC/equations.gms:296:!! Assuming curtailment has a triangle shape, 1x wide, 3x high => 3x^2 would be the area of the rectangle. The area of the triangle = 1/2 times this.
modules/32_power/RLDC/equations.gms:297:!! Area = 3/2 x^2 = curt.   ==> x = 2/3 curt ^ 0.5
modules/32_power/RLDC/equations.gms:298-

core/presolve.gms:43-    * ( ( 1000
core/presolve.gms:44:          !! use default per-capita investments if no investment data in gdx
core/presolve.gms:45:          !! (due to different region settings)
core/presolve.gms:46-        * ( (p_inv_gdx("2005",regi) / pm_pop("2005",regi))$( p_inv_gdx("2005",regi) )
--
core/presolve.gms:201-
core/presolve.gms:202:  !! The demand reduction function a = 160 / (p + 200) + 0.2 assumes that demand 
core/presolve.gms:203:  !!  for cement is reduced by 40% if the price doubles (CO2 price of $200) and
core/presolve.gms:204:  !!  that demand reductions of 80% can be achieved in the limit.
core/presolve.gms:205-  pm_ResidualCementDemand("2005",regi) = 1;
--
core/presolve.gms:210-
core/presolve.gms:211:  !! Demand can only be reduced by 1% p.a.
core/presolve.gms:212-  loop (ttot$( ttot.val gt 2005 ),
--
core/presolve.gms:229-
core/presolve.gms:230:  !! Costs of cement demand reduction are the integral under the activity 
core/presolve.gms:231:  !! reduction curve times baseline emissions.
core/presolve.gms:232:  !! a = 160 / (p + 200) + 0.2
core/presolve.gms:233:  !! A = 160 ln(p + 200) + 0.2p
core/presolve.gms:234:  !! A_MAC(p*) = A(p*) - A(0) - a(p*)p*
core/presolve.gms:235-  pm_CementDemandReductionCost(ttot,regi)$( ttot.val ge 2005 )

core/datainput.gms:140-
core/datainput.gms:141:!! Modify spv and storspv parameters for optimistic VRE supply assumptions
core/datainput.gms:142-if (cm_VRE_supply_assumptions eq 1,
--
core/datainput.gms:1213-loop(te$( teNoLearn(te) AND teRegTechCosts(te) ),
core/datainput.gms:1214:  !! no value after 2020 is currently used (see convergence below)
core/datainput.gms:1215-  pm_inco0_t(ttot,regi,te)$( ttot.val ge 2015 AND ttot.val lt 2040 )

core/loop.gms:23-$ifthen.subsectors "%industry%" == "subsectors"       !! industry
core/loop.gms:24:!! Calibrating industry/subsectors lead to random infeasibilities on the order
core/loop.gms:25:!! of 1e-15.  Relaxing this attribute a little solves this problem.
core/loop.gms:26-hybrid.tolinfeas = 1e-14;
--
core/loop.gms:160-  execute_unload 'fulldata';
core/loop.gms:161:  !! retain gdxes of intermediate iterations by copying them using shell
core/loop.gms:162:  !! commands
core/loop.gms:163-  if (c_keep_iteration_gdxes eq 1,

core/postsolve.gms:306-        (pm_ts(ttot2) / 2 + 0.5)$( ttot2.val eq 2020 )
core/postsolve.gms:307:        !! entire middle periods
core/postsolve.gms:308-      + (pm_ts(ttot2))$( 2020 lt ttot2.val AND ttot2.val lt ttot.val )

core/sets.gms:403-
core/sets.gms:404:        !! materials, feedstock, and industrial goods
core/sets.gms:405-        steel        "Steel"
--
core/sets.gms:409-
core/sets.gms:410:        !! emissions
core/sets.gms:411-        co2          "carbon dioxide emissions"
--
core/sets.gms:453-
core/sets.gms:454:        !! emissions from industry sub-sectors
core/sets.gms:455-        co2cement      "CO2 emissions from clinker and cement production"
--
core/sets.gms:459-
core/sets.gms:460:        !! various emissions
core/sets.gms:461-        emiFgasTotal    "XXX"
--
core/sets.gms:642-
core/sets.gms:643:  !! production factors of industry with subsectors
core/sets.gms:644-  ue_industry             "useful energy of industry sector"
--
core/sets.gms:1650-
core/sets.gms:1651:        !! emissions from industry sub-sectors
core/sets.gms:1652-        co2cement      "CO2 emissions from clinker and cement production"

core/preloop.gms:146-
core/preloop.gms:147:!! all net negative co2luc
core/preloop.gms:148-p_macBaseMagpieNegCo2(t,regi) = pm_macBaseMagpie(t,regi,"co2luc")$(pm_macBaseMagpie(t,regi,"co2luc") < 0);

core/equations.gms:60-  * (1.02 + pm_prtp(regi) ) ** (pm_ts(t) / 2) !! This increases the investments as if the money was actually borrowed 
core/equations.gms:61:  !! half a time step earlier, using an interest rate of pm_prtp + 2%, which is close to the model-endogenous interest rate. 
core/equations.gms:62:  !! We do this to reduce the difference to the previous version where the effect of deltacap on capacity was split 
core/equations.gms:63:  !! half to the current and half to the next time.   
core/equations.gms:64-;
--
core/equations.gms:77-  * (1.02 + pm_prtp(regi) ) ** (pm_ts(t) / 2) !! This increases the investments as if the money was actually borrowed 
core/equations.gms:78:  !! half a time step earlier, using an interest rate of pm_prtp + 2%, which is close to the model-endogenous interest rate. 
core/equations.gms:79:  !! We do this to reduce the difference to the previous version where the effect of deltacap on capacity was split 
core/equations.gms:80:  !! half to the current and half to the next time. 
core/equations.gms:81-;
--
core/equations.gms:310-  =e=
core/equations.gms:311:!! early retirement for some fossil technologies
core/equations.gms:312-  (1 - vm_capEarlyReti(ttot,regi,te))
--
core/equations.gms:688-  + sum(teCCS2rlf(te,rlf), vm_ccs_cdr(t,regi,"cco2","ico2","ccsinje",rlf)))
core/equations.gms:689:  !! scaled by the fraction that gets stored geologically
core/equations.gms:690-  * (sum(teCCS2rlf(te,rlf),
--
core/equations.gms:693-        vm_co2capture(t,regi,"cco2","ico2","ccsinje",rlf))+sm_eps))
core/equations.gms:694:  !! net negative emissions from co2luc
core/equations.gms:695-  -  p_macBaseMagpieNegCo2(t,regi)
core/equations.gms:696:       !! negative emissions from the cdr module that are not stored geologically
core/equations.gms:697-       -       (vm_emiCdr(t,regi,"co2") + sum(teCCS2rlf(te,rlf), vm_ccs_cdr(t,regi,"cco2","ico2","ccsinje",rlf)))
```

Reply:
There is no argument against evidences. You can ignore my suggestion then.